src/screens/DiscoveryScreenV2/index.js

```diff
@@ -92,26 +92,31 @@ const DiscoveryScreenV2 = ({route}) => {
     };
 
     const fetchDiscoveryDataUser = async () => {
-      const cancelToken = cancelTokenRef?.current?.token;
-      const data = await DiscoveryRepo.fetchDiscoveryDataUser(text, false, {cancelToken});
-      if (data.success) {
-        setDiscoveryDataFollowedUsers(
-          data?.followedUsers?.map((item) => ({
-            ...item,
-            following: item.user_id_follower !== null
-          })) || []
-        );
-        setDiscoveryDataUnfollowedUsers(
-          data?.unfollowedUsers?.map((item) => ({
-            ...item,
-            following: item.user_id_follower !== null
-          })) || []
-        );
+      try {
+        const cancelToken = cancelTokenRef?.current?.token;
+        const data = await DiscoveryRepo.fetchDiscoveryDataUser(text, false, {cancelToken});
+        if (data.success) {
+          setDiscoveryDataFollowedUsers(
+            data?.followedUsers?.map((item) => ({
+              ...item,
+              following: item.user_id_follower !== null
+            })) || []
+          );
+          setDiscoveryDataUnfollowedUsers(
+            data?.unfollowedUsers?.map((item) => ({
+              ...item,
+              following: item.user_id_follower !== null
+            })) || []
+          );
+        }
+      } catch (e) {
+        console.log('e', e);
+      } finally {
+        setIsLoadingDiscovery((prevState) => ({
+          ...prevState,
+          user: false
+        }));
       }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling has been added to the `fetchDiscoveryDataUser` function with a try-catch block. This is a good practice as it prevents the application from crashing in case of an unexpected error during the execution of the async function. However, it would be better to handle the error more gracefully than just logging it to the console. Consider dispatching an action to update the state with the error message or showing a user-friendly error message.
```diff
- const fetchDiscoveryDataUser = async () => {
+ const fetchDiscoveryDataUser = async () => {
+   try {
      const cancelToken = cancelTokenRef?.current?.token;
      const data = await DiscoveryRepo.fetchDiscoveryDataUser(text, false, {cancelToken});
      if (data.success) {
        setDiscoveryDataFollowedUsers(
          data?.followedUsers?.map((item) => ({
            ...item,
            following: item.user_id_follower !== null
          })) || []
        );
        setDiscoveryDataUnfollowedUsers(
          data?.unfollowedUsers?.map((item) => ({
            ...item,
            following: item.user_id_follower !== null
          })) || []
        );
       }
      setIsLoadingDiscovery((prevState) => ({
        ...prevState,
        user: false
      }));
+   } catch (e) {
+     console.log('e', e);
+   }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->